### PR TITLE
Add Outlook Reply By Email Feature

### LINKED ISSUE #3426: Expand reply-by-email feature For outlook
```We are preparing to participate in Google Code-in, and have reserved this issue for participants in GCI - but we'd love to have your help with another one! Please check out https://code.publiclab.org to see more.```

Currently in `reply-by-email` feature we support parsing of email for google, yahoo and the domains that are using google service.
And as this feature is working stable, we can add the feature of parsing outlook mail so that `reply-by-email` support one more domain.
To implement this feature, starting point is https://github.com/publiclab/plots2/blob/master/app/models/comment.rb#L261

### PR DESCRIPTION
**Pull request by kevinzluo**
Fixes #3426  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

I have added outlook reply by email. The parsing is done using regex instead of nokogiri.
ANSWERS:
Replying once:
showfirstoutlookrevamp

Replying twice:
showsecondoutlookrevamp

NOTES/QUESTIONS:
Replying Once:
showfirstoutlooknoterevamp

Replying Twice:
showsecondoutlooknoterevamp

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [X] code is in uniquely-named feature branch and has no merge conflicts
* [X] PR is descriptively titled
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by kevinzluo** [comment]
@namangupta01 Could you take a look? I couldn't figure out how to do it with css selectors in Nokogiri so I ended up just using Regex.

**Comment by kevinzluo** [comment]
@publiclab/reviewers Would it be fine for me to continue writing the unit and integration tests on this PR?

**Comment by kevinzluo** [comment]
@SidharthBansal Could you take a look at this as well?

**Comment by SidharthBansal** [comment]
@souravirus or @sagarpreet or @jywarren or @namangupta01 or Gaurav can help.
Please tag them if I have done any spelling mistake in their names.
Sorry for that. I am in train and helping you guys on the mobile.

**Comment by kevinzluo** [comment]
Thank you! @sagarpreet-chadha @gauravano Could you take a look?

File: app/models/comment.rb [hunk 1]
```diff
@@ -332,6 +336,24 @@ def self.gmail_parsed_mail(mail_doc)
...
+      extra_content = nil
+    else
+      body_message = separator[1].match(/(.+)(<body dir="ltr">)(.+)/m)
+      comment_content = Nokogiri::HTML(body_message[3])
+      trimmed_message = separator[3].match(/(.+)(<\/body>)(.+)/m)
+      extra_content = Nokogiri::HTML(trimmed_message[1])
+    end
+
+    {
+      "comment_content" => comment_content,
```
Comment (Souravirus) [inline app/models/comment.rb]: Please write these in this notation: `comment_content: comment_content`.

Comment (kevinzluo) [reply app/models/comment.rb]: Would you like me to change it for the other parsing methods as well?

Comment (Souravirus) [reply app/models/comment.rb]: Yeah please change those also.

Comment (kevinzluo) [reply app/models/comment.rb]: @Souravirus I have made the changes. Can I also write the integration tests on this PR?

Comment (Souravirus) [reply app/models/comment.rb]: Yeah nice work. You can write the integration tests related to this if you have time to do so. Thank you!!

Comment (kevinzluo) [reply app/models/comment.rb]: I've been trying to write the integration tests but I keep encountering this issue:
image

I have added my user to the users fixture and can't seem to figure out why this happens.

**Comment by kevinzluo** [comment]
@SidharthBansal This was one of the tasks I completed but didn't claim on the GCI dashboard. 

Thank you for all your work!
Kevin Luo

**Comment by SidharthBansal** [comment]
You need not worry. We will be judging you all on quality and quantity both. We will be judging all the conversations, collaborative actions to other people etc.
There are a wide variety of parameters upon which you will be judged, so don't worry if you forget to get points on GCI dashboard.
:-)

**Comment by kevinzluo** [comment]
Thank you for your reply @SidharthBansal. Could we wait for me to finish the integration tests? That way we can ensure that it is working.

Also, I believe I have found an error with automated email sending to users. Sometimes when a person posts a new comment the update email sent to the admin uses the note id number but puts an "a" in front of it. I will raise an issue once I get home.

**Comment by kevinzluo** [comment]
@publiclab/reviewers 
Could you take a look at the integration test? 
Also, do I need to copy this test into `token_comment_test.rb`, because it seems the other two integration tests are duplicated there also.

File: app/models/comment.rb (hunk 1 above)
Comment (kevinzluo) [reply app/models/comment.rb]: @Souravirus could you take a look? I think I have finished the tests.

**Comment by namangupta01** [comment]
Nice Work @kevinzluo 👍

**Comment by namangupta01** [comment]
Congrats @kevinzluo!
Very well implemented 🎉

**Comment by kevinzluo** [comment]
Thank you!

**Comment by namangupta01** [comment]
Hey @jywarren, this is good to merge!

**Comment by gauravano** [comment]
@jywarren this one is big. You might want to merge this one yourself. Thanks!

**Comment by jywarren** [comment]
This is really nicely done. Great teamwork everyone and many thanks, @kevinzluo !!

**Comment by jywarren** [comment]
With all you've learned about this system here, would you perhaps be interested in trying to catch this elusive bug in hiding quoted replies from Google Groups? 

https://github.com/publiclab/plots2/issues/3218#issuecomment-447607768

We'd love your help if so! Thanks again!!!

**Pull request merged**